src/primitive_flash_registry.cc

```diff
@@ -275,35 +328,111 @@ PRIMITIVE(region_read) {
   ARGS(FlashRegion, resource, int, from, MutableBlob, bytes);
   int size = bytes.length();
   if (!is_within_bounds(resource, from, size)) OUT_OF_BOUNDS;
-  FlashRegistry::flush();
-  const uint8* region = FlashRegistry::region(resource->offset(), resource->size());
-  memcpy(bytes.address(), region + from, size);
+  word offset = resource->offset();
```

Comment:
This needs a proper refactoring. The current flash registry works with offsets relative to a specific partition and it is special because it is mmap'ed in (because we want the interpreter to be able to use the automatic IRAM cache).